src/gui/addnewtorrentdialog.cpp

```diff
@@ -94,6 +95,8 @@ AddNewTorrentDialog::AddNewTorrentDialog(QWidget *parent)
     connect(ui->browseButton, SIGNAL(clicked()), SLOT(browseButton_clicked()));
     ui->defaultSavePathCheckBox->setVisible(false); // Default path is selected by default
 
+    ui->doNotDeleteTorrentCheckBox->setVisible(TorrentFileGuard::deleteAddedTorrents());
```

Comment:
Sorry, I'm lost. The checkbox in this dialog (Add Torrent) does not belong to any group. Do you suggest to make it visible always, replacing the `setVisible()` call with `setEnbaled()`?